### PR TITLE
Add alidns provider configs for globaldns

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/17838

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

**Comment by orangedeng** [comment]
I think this PR should be merged after we push `registry.opensource.zalan.do/teapot/external-dns` to `rancher/kubernetes-external-dns`.

File: apis/management.cattle.io/v3/k8s_defaults.go [hunk 1]
```diff
@@ -109,6 +110,9 @@ var (
 			FluentdHelper:                 m("rancher/fluentd-helper:v0.1.2"),
 			LogAggregatorFlexVolumeDriver: m("rancher/log-aggregator:v0.1.4"),
 		},
+		GlobalDNSSystemImages: GlobalDNSSystemImages{
+			ExternalDNS: m("registry.opensource.zalan.do/teapot/external-dns:v0.5.10"),
```
Comment (alena1108) [inline apis/management.cattle.io/v3/k8s_defaults.go]: @gitlawr @prachidamle we should switch to using tagged version of external dns

**Comment by alena1108** [comment]
@gitlawr  `rancher/kubernetes-external-dns:v0.5.10` is up

**Pull request merged**